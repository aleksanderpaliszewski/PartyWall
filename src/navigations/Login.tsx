import React from 'react';
import {Scenes} from '../api/enums';
import {
  createStackNavigator,
  StackNavigationProp,
} from '@react-navigation/stack';
import {RouteProp} from '@react-navigation/native';
import Login from '../scenes/Login';
import Register from '../scenes/Register';

export type ScreenName = Scenes.Login | Scenes.Register;

export type StackParamList = {
  [Scenes.Login]: undefined;
  [Scenes.Register]: undefined;
};

export interface LoginStackScreenProps<T extends ScreenName> {
  navigation: StackNavigationProp<StackParamList, T>;
  route: RouteProp<StackParamList, T>;
}

const {Navigator, Screen} = createStackNavigator<StackParamList>();

const LoginStackScreen = () => {
  return (
    <Navigator
      screenOptions={{
        headerShown: false,
      }}>
      <Screen name={Scenes.Login} component={Login} />
      <Screen name={Scenes.Register} component={Register} />
    </Navigator>
  );
};

export default LoginStackScreen;
