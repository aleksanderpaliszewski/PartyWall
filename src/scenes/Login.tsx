import React, {FC, useState, useContext} from 'react';
import {StyleSheet, View} from 'react-native';
import {FormikValues} from 'formik';

import Layout from '../components/Layout';
import LoginForm from '../components/LoginForm';
import ApiContext from '../utils/apiContext';
import UserContext from '../utils/userContext';
import SnackBarContext from '../utils/snackbarContext';

import {User} from '../api/interface';

const Login: FC = () => {
  const api = useContext(ApiContext);
  const {setAuthData} = useContext(UserContext);
  const {setMessage} = useContext(SnackBarContext);

  const [loading, setLoading] = useState(false);

  const handleSubmit = ({email, password}: FormikValues) => {
    setLoading(true);

    return api
      .post<User>('auth/login', {email, password})
      .then(({data}) => setAuthData(data))
      .catch((error) => setMessage(error.message))
      .finally(() => setLoading(false));
  };

  return (
    <Layout>
      <View style={styles.container}>
        <LoginForm handleSubmit={handleSubmit} loading={loading} />
      </View>
    </Layout>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'flex-end',
  },
});

export default Login;
