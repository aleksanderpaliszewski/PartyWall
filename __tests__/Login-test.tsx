import React from 'react';
import Login from '../src/scenes/Login';
import renderer from 'react-test-renderer';
import {getLoginProps} from './navigation';

describe('login', () => {
  it('should render without crashing', () => {
    const mockProps = getLoginProps();

    const rendered = renderer.create(<Login {...mockProps} />).toJSON();
    expect(rendered).toBeTruthy();
  });

  it('should render correctly', () => {
    const mockProps = getLoginProps();

    const rendered = renderer.create(<Login {...mockProps} />).toJSON();
    expect(rendered).toMatchSnapshot();
  });
});
