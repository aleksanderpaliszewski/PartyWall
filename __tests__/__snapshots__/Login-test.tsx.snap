// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login should render correctly 1`] = `
<BVLinearGradient
  colors={
    Array [
      4281343166,
      4285726784,
    ]
  }
  endPoint={
    Object {
      "x": 1,
      "y": 1,
    }
  }
  locations={null}
  startPoint={
    Object {
      "x": 0,
      "y": 0,
    }
  }
  style={
    Object {
      "flex": 1,
      "opacity": 0.8,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Array [
          Object {
            "flex": 1,
            "justifyContent": "space-between",
          },
          Object {
            "marginTop": 0,
          },
        ],
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "flex-end",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "borderTopLeftRadius": 40,
            "borderTopRightRadius": 40,
            "opacity": 0.95,
            "paddingHorizontal": 30,
            "paddingVertical": 60,
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "elevation": 2,
              "marginVertical": 10,
              "shadowColor": "#000000",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.1,
              "shadowRadius": 2,
            }
          }
        >
          <View>
            <View
              pointerEvents="none"
              style={
                Array [
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 6,
                  },
                  Object {
                    "backgroundColor": "#FFFFFF",
                    "borderColor": "#dcdcdc",
                    "borderRadius": 7,
                    "borderWidth": 1,
                  },
                ]
              }
            />
            <View
              style={
                Object {
                  "minHeight": 50,
                  "paddingBottom": 0,
                  "paddingTop": 8,
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    Object {
                      "backgroundColor": "#FFFFFF",
                      "bottom": 0,
                      "height": 2,
                      "left": 10,
                      "opacity": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 6,
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                      ],
                      "width": 8,
                    }
                  }
                />
                <Text
                  numberOfLines={1}
                  style={
                    Object {
                      "backgroundColor": "#FFFFFF",
                      "color": "transparent",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "left": 18,
                      "opacity": 0,
                      "paddingHorizontal": 0,
                      "position": "absolute",
                      "textAlign": "left",
                      "top": 34,
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                        Object {
                          "translateY": 0,
                        },
                        Object {
                          "scale": 1,
                        },
                        Object {
                          "scaleY": 1,
                        },
                      ],
                      "writingDirection": "ltr",
                    }
                  }
                >
                  E-mail
                </Text>
                <Text
                  numberOfLines={1}
                  onLayout={[Function]}
                  style={
                    Object {
                      "color": "#6200ee",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "left": 0,
                      "opacity": 0,
                      "paddingHorizontal": 14,
                      "position": "absolute",
                      "textAlign": "left",
                      "top": 33,
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                        Object {
                          "translateY": 0,
                        },
                        Object {
                          "scale": 1,
                        },
                      ],
                      "writingDirection": "ltr",
                    }
                  }
                >
                  E-mail
                </Text>
                <Text
                  numberOfLines={1}
                  style={
                    Object {
                      "color": "#dcdcdc",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "left": 0,
                      "opacity": 0,
                      "paddingHorizontal": 14,
                      "position": "absolute",
                      "textAlign": "left",
                      "top": 33,
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                        Object {
                          "translateY": 0,
                        },
                        Object {
                          "scale": 1,
                        },
                      ],
                      "writingDirection": "ltr",
                    }
                  }
                >
                  E-mail
                </Text>
              </View>
              <TextInput
                allowFontScaling={true}
                autoCapitalize="none"
                editable={true}
                fieldName="email"
                multiline={false}
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                placeholder=""
                placeholderTextColor="#730040"
                rejectResponderTermination={true}
                selectionColor="#6200ee"
                style={
                  Array [
                    Object {
                      "flexGrow": 1,
                      "margin": 0,
                      "paddingHorizontal": 14,
                      "zIndex": 1,
                    },
                    Object {
                      "height": 50,
                    },
                    Object {
                      "paddingBottom": 17,
                      "paddingTop": 17,
                    },
                    Object {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "textAlign": "left",
                      "textAlignVertical": "center",
                    },
                    Array [
                      Object {},
                    ],
                  ]
                }
                testID="email"
                textContentType="emailAddress"
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "elevation": 2,
              "marginVertical": 10,
              "shadowColor": "#000000",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.1,
              "shadowRadius": 2,
            }
          }
        >
          <View>
            <View
              pointerEvents="none"
              style={
                Array [
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 6,
                  },
                  Object {
                    "backgroundColor": "#FFFFFF",
                    "borderColor": "#dcdcdc",
                    "borderRadius": 7,
                    "borderWidth": 1,
                  },
                ]
              }
            />
            <View
              style={
                Object {
                  "minHeight": 50,
                  "paddingBottom": 0,
                  "paddingTop": 8,
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    Object {
                      "backgroundColor": "#FFFFFF",
                      "bottom": 0,
                      "height": 2,
                      "left": 10,
                      "opacity": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 6,
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                      ],
                      "width": 8,
                    }
                  }
                />
                <Text
                  numberOfLines={1}
                  style={
                    Object {
                      "backgroundColor": "#FFFFFF",
                      "color": "transparent",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "left": 18,
                      "opacity": 0,
                      "paddingHorizontal": 0,
                      "position": "absolute",
                      "textAlign": "left",
                      "top": 34,
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                        Object {
                          "translateY": 0,
                        },
                        Object {
                          "scale": 1,
                        },
                        Object {
                          "scaleY": 1,
                        },
                      ],
                      "writingDirection": "ltr",
                    }
                  }
                >
                  Password
                </Text>
                <Text
                  numberOfLines={1}
                  onLayout={[Function]}
                  style={
                    Object {
                      "color": "#6200ee",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "left": 0,
                      "opacity": 0,
                      "paddingHorizontal": 14,
                      "position": "absolute",
                      "textAlign": "left",
                      "top": 33,
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                        Object {
                          "translateY": 0,
                        },
                        Object {
                          "scale": 1,
                        },
                      ],
                      "writingDirection": "ltr",
                    }
                  }
                >
                  Password
                </Text>
                <Text
                  numberOfLines={1}
                  style={
                    Object {
                      "color": "#dcdcdc",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "left": 0,
                      "opacity": 0,
                      "paddingHorizontal": 14,
                      "position": "absolute",
                      "textAlign": "left",
                      "top": 33,
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                        Object {
                          "translateY": 0,
                        },
                        Object {
                          "scale": 1,
                        },
                      ],
                      "writingDirection": "ltr",
                    }
                  }
                >
                  Password
                </Text>
              </View>
              <TextInput
                allowFontScaling={true}
                editable={true}
                fieldName="password"
                multiline={false}
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                placeholder=""
                placeholderTextColor="#730040"
                rejectResponderTermination={true}
                secureTextEntry={true}
                selectionColor="#6200ee"
                style={
                  Array [
                    Object {
                      "flexGrow": 1,
                      "margin": 0,
                      "paddingHorizontal": 14,
                      "zIndex": 1,
                    },
                    Object {
                      "height": 50,
                    },
                    Object {
                      "paddingBottom": 17,
                      "paddingTop": 17,
                    },
                    Object {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "textAlign": "left",
                      "textAlignVertical": "center",
                    },
                    Array [
                      Object {},
                    ],
                  ]
                }
                testID="password"
                textContentType="password"
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "#6200ee",
              "borderColor": "transparent",
              "borderRadius": 30,
              "borderStyle": "solid",
              "borderWidth": 0,
              "elevation": 2,
              "marginTop": 30,
              "minWidth": 64,
              "shadowColor": "#000000",
              "shadowOffset": Object {
                "height": 0.75,
                "width": 0,
              },
              "shadowOpacity": 0.24,
              "shadowRadius": 1.5,
              "width": 100,
            }
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              Object {
                "disabled": false,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 30,
                },
              ]
            }
            testID="button"
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  },
                  undefined,
                ]
              }
            >
              <Text
                numberOfLines={1}
                selectable={false}
                style={
                  Array [
                    Object {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontWeight": "400",
                      "textAlign": "left",
                    },
                    Array [
                      Object {
                        "letterSpacing": 1,
                        "marginHorizontal": 16,
                        "marginVertical": 9,
                        "textAlign": "center",
                      },
                      undefined,
                      Object {
                        "textTransform": "uppercase",
                      },
                      Object {
                        "color": "#ffffff",
                        "fontFamily": "System",
                        "fontWeight": "500",
                      },
                      Object {
                        "fontFamily": "System",
                        "fontWeight": "500",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                Log in
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</BVLinearGradient>
`;
